### ISSUE TITLE
okio integration

### ISSUE THREAD

**Issue by vanniktech** [tamimattafi/krop#30]
On top of https://github.com/tamimattafi/krop/pull/24 we could add a new module for an integration with okio such that we can also pass in an okio `Source` as an image source. Obviously, this should be in a separate module.

**Comment by tamimattafi** [comment]
@vanniktech As I thought about okio and kotlin.io integration, I came to a conclusion that if we start supporting each solution separately, we will have many modules. Some people will want to integrate other ViewModel implementations, or other navigation tools, or even DI. This might transform a lightweight library to a complicated framework-like solution.

What do you think about abstraction for such cases? We can give users a default implementation that doesn't depend on many third-party libraries and solely on the platform. Also, we give users the freedom of implementing their own solutions for different features like saving and importing pictures.

As for material3, we can do something similar by abstracting some UI components and providing a default implementation for material. But for other design languages like Cupertino, we shouldn't add extra modules that need active maintenance, but rather give users the freedom to implement their own style and components.

What do you think?

**Comment by vanniktech** [comment]
I agree. I also don't know either where to draw the line between having too many modules supporting the popular libraries and having no official support for them. Also the more modules we add, the more complex the documentation gets.

**Comment by tamimattafi** [comment]
@vanniktech I will think more about this. I would appreciate any suggestions.
